tests/test_message.py

```diff
@@ -6,47 +6,57 @@
 
 def test_basic_url_detection_of_https():
     expected = "https://example.org"
-    assert message.extract_url("foo <https://example.org>") == expected
+    text = "foo <https://example.org>"
+    assert message.extract_url(text) == expected
 
 def test_hash_in_url():
-    expected = "https://#.org"
-    assert message.extract_url("<https://#.org>") == expected
+    expected = "https://foo.org#bar"
+    text = "<https://foo.org#bar>"
+    assert message.extract_url(text) == expected
 
 
 def test_plus_in_url():
-    expected = "https://+.org"
-    assert message.extract_url("<https://+.org>") == expected
+    expected = "http://foo.org/foo+bar"
+    text = "<http://foo.org/foo+bar>"
+    assert message.extract_url(text) == expected
 
 
 def test_equal_in_url():
-    expected = "https://=.org"
-    assert message.extract_url("<https://=.org>") == expected
+    expected = "https://wikipedia.org/title=foo"
+    text = "<https://wikipedia.org/title=foo>"
```

Comment:
I saw that but wondered if it made sense to have a question mark on the same one